src/item_contents.h

```diff
@@ -0,0 +1,87 @@
+#pragma once
+#ifndef ITEM_CONTENTS_H
+#define ITEM_CONTENTS_H
+
+#include "point.h"
+#include "ret_val.h"
+#include "type_id.h"
+#include "units.h"
+#include "visitable.h"
+
+#include <list>
+
+class Character;
+class item;
+
+using itype_id = std::string;
+
+class item_contents
+{
+    public:
+        item_contents() = default;
+        // used to aid migration
+        item_contents( const std::list<item> &items ) : items( items ) {}
+
+        bool empty() const;
+
+        // returns a list of pointers to all top-level items
```

Comment:
I'd suggest using `/** comment here */` for these comments so that doxygen picks them up. For multiline comments use the following:
```
/**
 * comment here
 * more comment
 */
```